x-pack/plugins/observability_onboarding/public/components/app/custom_logs/wizard/configure_logs.tsx

```diff
@@ -85,17 +141,30 @@ export function ConfigureLogs() {
     );
 
     if (index === 0) {
+      setIntegrationName(getFilename(filepath));
       setDatasetName(getFilename(filepath));
     }
   }
-
   const isDatasetNameInvalid = datasetNameTouched && isEmpty(datasetName);
-
+  const isIntegrationNameInvalid =
+    (integrationNameTouched && isEmpty(integrationName)) ||
+    (integrationError && integrationError.type === 'NamingCollision');
   const datasetNameError = i18n.translate(
     'xpack.observability_onboarding.configureLogs.dataset.error',
     { defaultMessage: 'A dataset name is required.' }
   );
 
+  const integrationNameError =
+    integrationError && integrationError.type === 'NamingCollision'
+      ? i18n.translate(
+          'xpack.observability_onboarding.configureLogs.integration.namingCollisionError',
+          { defaultMessage: 'An integration with this name already exists.' }
+        )
+      : i18n.translate(
+          'xpack.observability_onboarding.configureLogs.integration.emptyError',
+          { defaultMessage: 'An integration name is required.' }
+        );
+
```

Comment:
You could add lowercase validation to prevent the back and forth with the server by adding a condition here for the error validation:
```ts
const isLowerCase = (str: string) => str.toLowerCase() === str;

  const hasNamingCollision =
    integrationError && integrationError.type === 'NamingCollision';

  const isDatasetNameInvalid =
    datasetNameTouched && (isEmpty(datasetName) || !isLowerCase(datasetName));
  const isIntegrationNameInvalid =
    (integrationNameTouched &&
      (isEmpty(integrationName) || !isLowerCase(integrationName))) ||
    hasNamingCollision;

  const getIntegrationNameError = () => {
    if (integrationNameTouched && isEmpty(integrationName)) {
      return i18n.translate(
        'xpack.observability_onboarding.configureLogs.integration.emptyError',
        { defaultMessage: 'An integration name is required.' }
      );
    }
    if (integrationNameTouched && !isLowerCase(integrationName)) {
      return i18n.translate(
        'xpack.observability_onboarding.configureLogs.integration.lowercaseError',
        { defaultMessage: 'An integration name should be lowercase.' }
      );
    }
    if (hasNamingCollision) {
      return i18n.translate(
        'xpack.observability_onboarding.configureLogs.integration.namingCollisionError',
        { defaultMessage: 'An integration with this name already exists.' }
      );
    }
  };

  const getDatasetNameError = () => {
    if (datasetNameTouched && isEmpty(datasetName)) {
      return i18n.translate(
        'xpack.observability_onboarding.configureLogs.dataset.error',
        { defaultMessage: 'A dataset name is required.' }
      );
    }
    if (datasetNameTouched && !isLowerCase(datasetName)) {
      return i18n.translate(
        'xpack.observability_onboarding.configureLogs.dataset.error',
        { defaultMessage: 'A dataset name should be lowercase.' }
      );
    }
  };

  const datasetNameError = getDatasetNameError();
  const integrationNameError = getIntegrationNameError();
```

We could add a more scalable way to validate this input using existing libraries, but this is the fastest change to handle these cases in the existing implementation, wdyt?

Reply:
Yeah, I can add the lowercase validation to the client side cases 👍

> We could add a more scalable way to validate this input using existing libraries, but this is the fastest change to handle these cases in the existing implementation, wdyt?

Yeah, I didn't want to get too invested in large refactors here, because the state management is definitely going to need to be enhanced when we get to the next stage of this work (I left a note in the PR description, but this will likely include enhancing state management and extracting the "create integration" resources (hooks, components etc) in to a reusable package).